projects/components/src/utils/test/locator/widget-locator.ts

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Copyright 2020 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { Type } from '@angular/core';
+
+/**
+ * A class that is able to make queries to the DOM and be instantiated by a widget finder.
+ *
+ * All widget locators should extend this base class.
+ *
+ * @example
+ *
+ * class LoginWidgetLocator<T> extends BaseWidgetLocator<T> {
+ *      getUsernameField(): T {
+ *          return this.driver.get('.username').unwrap();
+ *      }
+ *
+ *      getOkButton(): T {
+ *          return this.driver.getByText('button', 'Ok').unwrap();
+ *      }
+ *
+ *      getOkButtonContainer(): T {
+ *          return this.driver.getByText('button', 'Ok').parent('.button-container').unwrap();
+ *      }
+ * }
+ */
+export class BaseWidgetLocator<T> {
+    protected driver?: LocatorDriver<T>;
+    constructor(driver: LocatorDriver<T>) {
+        this.driver = driver;
+    }
+}
+
+/**
+ * Knows how to find elements of a given type T in the DOM.
+ */
+export interface LocatorDriver<T> {
+    /**
+     * Finds all child elements that match the given {@param cssSelector}.
+     */
+    get(cssSelector: string): LocatorDriver<T>;
+
+    /**
+     * Finds all child elements that match the given {@param cssSelector} and have text that contains the given {@param value}.
```

Comment:
I haven't heard of embedding `@param` within the description. They should be separate lines. Please add `@param` for each of the functions